pkg/services/provisioning/alerting/rules/provisioner.go

```diff
@@ -0,0 +1,163 @@
+package rules
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	alert_models "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/provisioning"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+type AlertRuleProvisioner interface {
+	Provision(ctx context.Context, path string) error
+}
+
+func NewAlertRuleProvisioner(
+	logger log.Logger,
+	dashboardService dashboards.DashboardService,
+	dashboardProvService dashboards.DashboardProvisioningService,
+	ruleService provisioning.AlertRuleService) AlertRuleProvisioner {
+	return &DefaultAlertRuleProvisioner{
+		logger:               logger,
+		cfgReader:            newRulesConfigReader(logger),
+		dashboardService:     dashboardService,
+		dashboardProvService: dashboardProvService,
+		ruleService:          ruleService,
+	}
+}
+
+type DefaultAlertRuleProvisioner struct {
+	logger               log.Logger
+	cfgReader            rulesConfigReader
+	dashboardService     dashboards.DashboardService
+	dashboardProvService dashboards.DashboardProvisioningService
+	ruleService          provisioning.AlertRuleService
+}
+
+func Provision(
+	ctx context.Context,
+	path string,
+	dashboardService dashboards.DashboardService,
+	dashboardProvisioningService dashboards.DashboardProvisioningService,
+	ruleService provisioning.AlertRuleService,
+) error {
+	ruleProvisioner := NewAlertRuleProvisioner(
+		log.New("provisioning"),
```

Comment:
We already created a logger with the same name in `provisioning.go`. Should we rename this, or just inject that logger?

Reply:
i changed this to `provisioning.alerting`, it's the same format as the others have.